File: src/com/dafttech/terra/game/world/tiles/TileSand.java
package com.dafttech.terra.game.world.tiles;

import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.dafttech.terra.game.world.World;
import com.dafttech.terra.resources.Resources;

public class TileSand extends TileFalling {

    public TileSand() {
        super();
    }

    @Override
    public TextureRegion getImage() {
        return Resources.TILES.getImage("sand");
    }

    @Override
    public float getFallSpeed(World world) {
        return 10;
    }

    @Override
    public float getFallDelay(World world) {
        return 0.2f;
    }
}

File: terra/src/main/scala/com/dafttech/terra/game/world/tiles/TileSand.scala
package com.dafttech.terra.game.world.tiles

import com.badlogic.gdx.graphics.g2d.TextureRegion
import com.dafttech.terra.game.world.GameWorld
import com.dafttech.terra.resources.Resources
import monix.eval.Task

class TileSand() extends TileFalling {
  override val getImage: Task[TextureRegion] = Resources.TILES.getImageTask("sand")

  override def getFallSpeed(world: GameWorld) = 20

  override def getFallDelay(world: GameWorld) = 0.2f
}